src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -229,6 +231,16 @@ int sd_dhcp_lease_get_captive_portal(sd_dhcp_lease *lease, const char **ret) {
         return 0;
 }
 
+int sd_dhcp_lease_get_dnr(sd_dhcp_lease *lease, ResolverData **ret_resolvers) {
+        assert_return(ret_resolvers, -EINVAL);
+
+        if (!lease->resolvers || lease->n_resolvers == 0)
```

Comment:
frankly, i'd just check n_resolvers. And maybe add an `assert(!!lease->resolvers == !!lease->n_resolvers)` here, after all if they two fields are either both null or neither null then there's a bug somewhere

Reply:
Since resolvers is a list and I don't use the length in very many places, I'm feeling like n_resolvers might just be redundant and I can calc the length anywhere I actually care. I'm probably iterating it anyway. I'll take another look at this.

Reply:
Removed this field.